stdlib/public/Cxx/std/String.swift

```diff
@@ -403,3 +403,66 @@ extension String {
     unsafe withExtendedLifetime(cxxU32StringView) {}
   }
 }
+
+@available(SwiftCompatibilitySpan 5.0, *)
+extension std.string {
+  public var span: Span<CChar> {
+    @_lifetime(borrow self)
+    @_alwaysEmitIntoClient
+    borrowing get {
+      let buffer = unsafe UnsafeBufferPointer(start: self.__dataUnsafe(), count: Int(self.size()))
+      let span = unsafe Span(_unsafeElements: buffer)
+      return unsafe _cxxOverrideLifetime(span, borrowing: self)
+    }
+  }
+}
+
+@available(SwiftStdlib 6.2, *)
+extension std.string {
+  public var utf8Span: UTF8Span? {
+    @_lifetime(borrow self)
+    @_alwaysEmitIntoClient
+    borrowing get {
+      let buffer = unsafe UnsafeBufferPointer(start: self.__dataUnsafe(), count: Int(self.size()))
+      let rawBuffer = UnsafeRawBufferPointer(buffer)
+      let bufferWithFixedType = unsafe rawBuffer.assumingMemoryBound(to: UInt8.self)
+      let span = unsafe Span(_unsafeElements: bufferWithFixedType)
+      let spanWithFixedLifetime = unsafe _cxxOverrideLifetime(span, borrowing: self)
+      do {
+        return try UTF8Span(validating: spanWithFixedLifetime)
```

Comment:
Couldn't you use `try?` and avoid the `do catch`?

Reply:
Yup, this works! Thanks for the tip!